backend/entities/tests/test_group.py

```diff
@@ -0,0 +1,235 @@
+    assert isinstance(group.creation_date, datetime)
+    assert group.terms_checked is True
+
+
+def test_required_fields() -> None:
+    """Test that required fields raise validation error when missing."""
+    user = UserFactory()
+    org = OrganizationFactory(created_by=user)
+
+    # Test missing group_name
+    with pytest.raises(ValidationError):
+        group = Group(
+            org_id=org,
+            created_by=user,
+            name="Test Name",
+            location="Test Location",
+            category="Test Category",
+        )
+        group.full_clean()
+
+    # Test missing location
+    with pytest.raises(ValidationError):
+        group = Group(
+            org_id=org,
+            created_by=user,
+            group_name="Test Group",
+            name="Test Name",
+            category="Test Category",
+        )
+        group.full_clean()
+
+
+def test_optional_fields() -> None:
+    """Test that optional fields can be blank or null."""
+    user = UserFactory()
+    org = OrganizationFactory(created_by=user)
+
+    group = Group.objects.create(
+        org_id=org,
+        created_by=user,
+        group_name="Test Group",
+        name="Test Name",
+        location="Test Location",
+        category="Test Category",
+    )
+
+    # Should not raise ValidationError
+    group.full_clean()
+    assert group.tagline == ""
+    assert group.get_involved_url == ""
+    assert group.terms_checked is False
+
+
+def test_field_max_lengths() -> None:
+    """Test that fields have correct max lengths."""
+    user = UserFactory()
+    org = OrganizationFactory(created_by=user)
+    fake = Faker()
+
+    group = Group.objects.create(
+        org_id=org,
+        created_by=user,
+        group_name=fake.company(),
+        name=fake.company(),
+        tagline=fake.catch_phrase(),
+        location=fake.city(),
+        category=fake.word(),
+        get_involved_url=fake.url(),
+        terms_checked=True,
+    )
+
+    assert len(group.group_name) <= 100
+    assert len(group.name) <= 100
+    assert len(group.tagline) <= 200
+    assert len(group.location) <= 100
+    assert len(group.category) <= 100
+    assert len(group.get_involved_url) <= 200
+
+
+def test_cascade_delete() -> None:
+    """Test that deleting an organization deletes all associated groups."""
+    user = UserFactory()
+    org = OrganizationFactory(created_by=user)
+
+    GroupFactory(org_id=org, created_by=user)
+
+    assert Group.objects.count() == 1
+    org.delete()
+    assert Group.objects.count() == 0
+
+    org = OrganizationFactory(created_by=user)
+    GroupFactory(org_id=org, created_by=user)
+
+    assert Group.objects.count() == 1
+    user.delete()
+    assert Group.objects.count() == 0
+
+
+def test_url_validations() -> None:
+    """Test that get_involved_url field is a valid URL."""
+    user = UserFactory()
+    org = OrganizationFactory(created_by=user)
+    fake = Faker()
+
+    # Test invalid URL
```

Comment:
Minor thing that we do for comments is that if they're their own line as is the case here, then it'd be best if they were fully punctuated as this is Python standard :) If it's an inline comment, then no need for capitalization or a period at the end, but if it's on its own line then including both would be best :) I can also do this if you'd like me to!